key_management/src/wallet.rs

```diff
@@ -336,4 +338,28 @@ mod tests {
         // check to make sure that the test_addr is actually the default addr for the wallet
         assert_eq!(wallet.get_default().unwrap(), test_addr);
     }
+
+    #[test]
+    fn secp_verify() {
+        let key_vec = construct_priv_keys();
+        let addr = key_vec[0].address.clone();
+        let mut wallet = Wallet::new_from_keys(MemKeyStore::new(), key_vec);
+
+        let msg = [0u8; 64];
+
+        let sig = wallet.sign(&addr, &msg).unwrap();
+        sig.verify(&msg, &addr).unwrap();
+    }
+
+    #[test]
+    fn bls_verify_test() {
+        let key_vec = construct_priv_keys();
+        let addr = key_vec[6].address.clone();
```

Comment:
instead of just indexing that vector, can you just generate each type explicitly, so that if construct_priv_keys changes for any reason, this doesn't break or just unknowingly lose coverage? Other option is just to assert the address protocol is Secp and BLS respectively?

Would be nice to also test the case of invalid address verify fails, as in try to verify with some other random address